python/lsst/ts/electrometer/controller.py

```diff
@@ -333,47 +360,115 @@ async def start_scan_dt(self, scan_duration):
 
     async def stop_scan(self):
         """Stop storing values in the Keithley electrometer."""
+        await self.get_intensity()
+        await self.csc.evt_intensity.set_write(intensity=self.last_value)
         self.log.debug("Stopping scan")
         self.manual_end_time = utils.current_tai()
+        self.scan_duration = self.manual_end_time - self.manual_start_time
         await self.send_command(f"{self.commands.stop_storing_buffer()}")
         await self.send_command(f"{self.commands.enable_display(True)}")
         self.log.debug("Scanning stopped, Now reading buffer.")
         res = await self.send_command(f"{self.commands.read_buffer()}", has_reply=True)
         intensity, times, temperature, unit, voltage = self.parse_buffer(res)
         await self.write_fits_file(intensity, times, temperature, unit, voltage)
 
-    async def write_fits_file(self, intensity, times, temperature, unit, voltage):
+    def make_primary_header(self):
+        """Make primary header for fits file that follows Rubin Obs. format."""
+        primary_hdu = fits.PrimaryHDU()
+        primary_hdu.header["FORMAT_V"] = ("1", "Header format version")
+        primary_hdu.header["OBSERVAT"] = "Vera C. Rubin Observatory"
+        primary_hdu.header["INSTRUME"] = (
+            f"Electrometer_index_{self.csc.salinfo.index}",
+            "Type of Instrument",
+        )
+        primary_hdu.header["MODEL"] = (self.model_id, "Model of instrument")
+        primary_hdu.header["LOCATN"] = (self.location, "Location of Instrument")
+        primary_hdu.header["ORIGIN"] = (
+            self.csc.salinfo.name,
+            "Name of the program that produced this data.",
+        )
+        primary_hdu.header["DATE-BEG"] = (
+            astropy.time.Time(self.manual_start_time, format="unix_tai").tai.fits,
```

Comment:
Please do not use astropy time. As Russell has explained many times, it is buggy. Use the ts_utils functions instead.